### PR TITLE
Openbsd ifdefs

### PR DESCRIPTION
**Pull request by bentley**
These fix most compilation problems under OpenBSD.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/4018)

### CODE REVIEW

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by Parlane** [comment]
This one fails our lint test.

See the output of this: https://buildbot.dolphin-emu.org/builders/lint/builds/2005/steps/shell/logs/stdio

**Comment by bentley** [comment]
Redid the last commit to wrap lines.

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Comment by Parlane** [comment]
Reviewed 5 of 6 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4018)*

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild you messed up!

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Comment by Parlane** [comment]
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/Core/HW/EXI_DeviceEthernet.h, line 330 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4018#-KMw6gF5d64Np2jP4iYO:-KMw6gF5d64Np2jP4iYP:1548813570) ([raw file](https://github.com/dolphin-emu/dolphin/blob/da43ecd24abeea260b1a2e6b043e1f7407bafd0b/Source/Core/Core/HW/EXI_DeviceEthernet.h#L330)):*
> ```objc
>   std::vector<u8> mWriteBuffer;
>   bool mWritePending;
> #elif defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) || \
> ```

Why dud you change these lines?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4018)*

**Comment by bentley** [comment]
I didn’t understand the wrapping rules in the linter. But that’s an old diff; cdf62dece39634367e9608db635c4f84c7a3b88c is the current one.

**Comment by phire** [comment]
The linter is `clang-format`, you can run it on your own computer and it will spit out a diff (if you install clang 3.8).

`git clang-format` will automatically apply the changes needed to the files staged for commit.

File: Source/Core/Core/HW/EXI_DeviceEthernet.h [hunk 1]
```diff
@@ -327,11 +327,12 @@ class CEXIETHERNET : public IEXIDevice
   OVERLAPPED mWriteOverlapped;
   std::vector<u8> mWriteBuffer;
   bool mWritePending;
-#elif defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)
+#elif defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__)
   int fd;
 #endif
 
-#if defined(WIN32) || defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(WIN32) || defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) ||          \
```
Comment (Parlane) [inline Source/Core/Core/HW/EXI_DeviceEthernet.h]: Does it really want all those spaces before the \ ?

**Comment by bentley** [comment]
*[Source/Core/Core/HW/EXI_DeviceEthernet.h, line 334 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4018#-KMsLgGzoePVGQagrjYp-r3-334:-KMwAFA2TP-_jSGvUHAi:-1775828933) ([raw file](https://github.com/dolphin-emu/dolphin/blob/cdf62dece39634367e9608db635c4f84c7a3b88c/Source/Core/Core/HW/EXI_DeviceEthernet.h#L334)):*
<details><summary><i>Previously, Parlane (Matthew Parlane) wrote…</i></summary>
> Does it really want all those spaces before the \ ?
</details>
That was straight from the output of the linter.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4018)*

**Comment by Parlane** [comment]
LGTM

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/4018#-:-KMwBh7AvjA805bUf7PV:676834233)*

**Pull request merged**